docs/selfhost/ngm-docker.md

```diff
@@ -0,0 +1,164 @@
+---
+sidebar_position: 5
+---
+
+# Using external Nginx Proxy Manager with Docker
+
+This Guide helps you host YAGPDB on Docker, with an external Nginx Proxy Manager.
+
+## Create and configure Discord Bot
+
+First, go to [https://discord.com/developers/applications](https://discord.com/developers/applications), then click on "New Application". 
+Type in your Bot Name, click on the little Checkbox and click on Create.
+
+Add your Icon if you want an Icon.
+
+Go to OAuth2, and follow these Steps:
+1. Click on Client ID: Copy (Save this for later)
+2. Click on Client Secret: Reset Secret (Save this for later)
+3. Click 2 Times on Add Redirect
+ 1. https://YOURHOST.COM/confirm_login
+ 2. https://YOURHOST.COM/manage
+
+Now go to Bot and follow these Steps:
+1. Click on Reset Token (Save this for later)
+2. Click on these Privileged Gateway Intents:
+ 1. Presence Intent
+ 2. Server Members Intent
+ 3. Message Content Intent
+
+## Install required packages
+
+```shell
+apt update
+apt install docker.io docker-compose git
+```
+
+## Clone Repository and copy neccasary Files
+
+```shell
+git clone https://github.com/botlabs-gg/yagpdb
+cp yagpdb/yagpdb_docker/{app.example.env,app.env}
+cp yagpdb/yagpdb_docker/{db.example.env,db.env}
+```
+
+## Edit app.py
+```shell
+cd yagpdb/yagpdb_docker
+nano app.env
+```
+Now edit these Settings:
+1. YAGPDB_OWNER: To your UserID (https://support.discordapp.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-)
+2. YAGPDB_CLIENTID: Your OAuth Client ID that you saved
+3. YAGPDB_CLIENTSECRET: Your OAuth Client Secret that you saved
+4. YAGPDB_BOTTOKEN: Your Bot Token that you saved
+5. YAGPDB_HOST + VIRTUAL_HOST + LETSENCRYPT_HOST: YOURHOST.COM
+6. Comment these Lines with # (Except you want the Plugins to be enabled):
+ 1. 36-39
+ 2. 42
+ 3. 51-54
+ 4. 57
+ 5. 61
+7. YAGPDB_EMAIL + LETSENCRYPT_EMAIL: Your Email
```

Comment:
This section is redundant with an earlier chapter and should be removed.
See https://hostyagpdb.xyz/discord-bot-setup/setupdiscordapplication/.